test/system/seed_deletion/delete_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the deletion of a seed
+ **/
+
+package seed_deletion
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/gardener/gardener/test/framework"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var seedName *string
+var shootName *string
+
+func init() {
+	seedName = flag.String("seed-name", "", "name of the seed")
+	shootName = flag.String("shoot-name", "", "name of the shoot")
+	framework.RegisterGardenerFrameworkFlags()
+}
+
+func validateFlags() {
+	if !framework.StringSet(*seedName) {
+		Fail("flag '-seed-name' needs to be specified")
+	}
+}
+
+var _ = Describe("Seed deletion testing", func() {
+
+	f := framework.NewGardenerFramework(nil)
+
+	framework.CIt("Testing if Seed can be deleted", func(ctx context.Context) {
```

Comment:
There's a bunch of code duplication in this function - deleting the seed and then the shoot and dumping the state, perhaps you can refactor this.